clarity/contracts/arkadiko-diko-init-v2-1.clar

```diff
@@ -0,0 +1,159 @@
+  (let (
+    (claimed-tokens (var-get foundation-tokens-claimed))
+    (pending-tokens (unwrap! (get-pending-foundation-tokens) ERR-NOT-AUTHORIZED))
+    (wallet (var-get foundation-wallet))
+  )
+    (asserts! (is-eq wallet tx-sender) ERR-NOT-AUTHORIZED)
+    (asserts! (>= pending-tokens amount) ERR-TOO-MANY-TOKENS-CLAIMED)
+    (var-set foundation-tokens-claimed (+ claimed-tokens amount))
+    (print { type: "foundation-token", action: "claimed", data: { amount: amount, recipient: wallet } })
+    (as-contract (contract-call? .arkadiko-token transfer amount (as-contract tx-sender) wallet none))
+  )
+)
+
+;; ---------------------------------------------------------
+;; Founders
+;; ---------------------------------------------------------
+
+;; @desc set founders (=team) wallet to given address
+;; @param address; new founders(=team) wallet
+;; @post boolean; returns a boolean indicating successfully set or not
+(define-public (set-founders-wallet (address principal))
+  (let (
+    (wallet (var-get founders-wallet))
+  )
+    (asserts! (is-eq wallet tx-sender) ERR-NOT-AUTHORIZED)
+    (var-set founders-wallet address)
+    (ok true)
+  )
+)
+
+;; Get number of founders tokens claimed already
+(define-read-only (get-claimed-founders-tokens)
+  (var-get founders-tokens-claimed)
+)
+
+;; Get amount of tokens founders can claim
+;; The founders are vested on 4 years, with a 6 months cliff.
+;; Vesting happens monthly. 21m / 48 months = 437.500 per month
+(define-read-only (get-pending-founders-tokens)
+  (let (
+    ;; Current month number after start
+    (month-number (/ (- burn-block-height (var-get contract-start-block)) BLOCKS-PER-MONTH))
+    (claimed-tokens (var-get founders-tokens-claimed))
+    (max-tokens-month (* month-number FOUNDERS-TOKENS-PER-MONTH))
+  )
+    ;; Vesting period
+    (if (and (>= month-number u6) (<= month-number u47))
+      (ok (- max-tokens-month claimed-tokens))
+      (if (> month-number u47)
+        ;; Vesting ended
+        (ok (- TOTAL-FOUNDERS claimed-tokens)) 
+        ;; Vesting did not start yet
+        (ok u0)
+      )
+    )
+  )
+)
+
+;; @desc claim tokens for team
+;; @param amount; amount of tokens to claim
+;; @post uint; returns amount of claimed tokens
+(define-public (founders-claim-tokens (amount uint))
+  (let (
+    (pending-tokens (unwrap! (get-pending-founders-tokens) ERR-NOT-AUTHORIZED))
+    (claimed-tokens (var-get founders-tokens-claimed))
+    (wallet (var-get founders-wallet))
+  )
+    (asserts! (is-eq wallet tx-sender) ERR-NOT-AUTHORIZED)
+    (asserts! (>= pending-tokens amount) ERR-TOO-MANY-TOKENS-CLAIMED)
+    (var-set founders-tokens-claimed (+ claimed-tokens amount))
+    (as-contract (contract-call? .arkadiko-token transfer amount (as-contract tx-sender) wallet none))
+  )
+)
+
+;; ---------------------------------------------------------
+;; Migrate
+;; ---------------------------------------------------------
+
+(define-public (migrate)
```

Comment:
Q: would it be better to execute this on deployment (i.e. just statements at the end of this file)? This way it is ran for sure and people know we wouldn't mint founder tokens twice etc.